lib/hocon/config_factory.rb

```diff
@@ -21,4 +22,25 @@ def self.parse_file_any_syntax(file_base_name, options)
   def self.empty(origin_description = nil)
     Hocon::Impl::ConfigImpl.empty_config(origin_description)
   end
+
+  def self.load_file(file_base_name, options_hash = nil)
```

Comment:
I think the options_hash could be pretty confusing for a user. I think we should change `load_file` method to take `parse_options` and `resolve_options` as optional parameters that default to `ConfigParseOptions.defaults` and `ConfigResolveOptions.defaults`, which can be used to specify none or both types of options, and we should add two new methods, `load_file_with_parse_options` and `load_file_with_resolve_options` that call into this method.